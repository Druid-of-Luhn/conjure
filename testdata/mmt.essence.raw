Spec
  { language = "Essence"
  , version = [ 2
              , 0
              ]
  , topLevelBindings = [ ( Find
                         , "mmt"
                         , DomainMatrix
                             { index = DomainIntegerFromTo (
                                         Just ( ValueInteger 1 ) ) (
                                         Just ( ValueInteger 3 ) )
                             , element = DomainMatrix
                                           { index = DomainIntegerFromTo (
                                                       Just ( ValueInteger 4 ) ) (
                                                       Just ( ValueInteger 6 ) )
                                           , element = DomainTuple
                                                         { components = [ DomainBoolean
                                                                        , DomainIntegerFromTo (
                                                                            Just ( ValueInteger 2 ) ) (
                                                                            Just ( ValueInteger 6 ) )
                                                                        ]
                                                         , representation = Nothing
                                                         }
                                           }
                             }
                         )
                       , ( Find
                         , "mmtm"
                         , DomainMatrix
                             { index = DomainIntegerFromTo (
                                         Just ( ValueInteger 1 ) ) (
                                         Just ( ValueInteger 3 ) )
                             , element = DomainMatrix
                                           { index = DomainIntegerFromTo (
                                                       Just ( ValueInteger 4 ) ) (
                                                       Just ( ValueInteger 6 ) )
                                           , element = DomainTuple
                                                         { components = [ DomainBoolean
                                                                        , DomainIntegerFromTo (
                                                                            Just ( ValueInteger 2 ) ) (
                                                                            Just ( ValueInteger 6 ) )
                                                                        , DomainMatrix
                                                                            { index = DomainIntegerFromTo (
                                                                                        Just ( ValueInteger 0 ) ) (
                                                                                        Just ( ValueInteger 9 ) )
                                                                            , element = DomainIntegerFromTo (
                                                                                          Just ( ValueInteger 10 ) ) (
                                                                                          Just ( ValueInteger 20 ) )
                                                                            }
                                                                        ]
                                                         , representation = Nothing
                                                         }
                                           }
                             }
                         )
                       ]
  , topLevelWheres = []
  , objective = Nothing
  , constraints = [ GenericNode
                      Eq
                      [ GenericNode
                          Index
                          [ GenericNode
                              Index
                              [ GenericNode
                                  Index
                                  [ Identifier "mmt"
                                  , ValueInteger 1
                                  ]
                              , ValueInteger 4
                              ]
                          , ValueInteger 0
                          ]
                      , ValueBoolean True
                      ]
                  , GenericNode
                      Eq
                      [ GenericNode
                          Index
                          [ GenericNode
                              Index
                              [ GenericNode
                                  Index
                                  [ Identifier "mmt"
                                  , ValueInteger 1
                                  ]
                              , ValueInteger 4
                              ]
                          , ValueInteger 1
                          ]
                      , ValueInteger 3
                      ]
                  , GenericNode
                      Eq
                      [ GenericNode
                          Index
                          [ GenericNode
                              Index
                              [ GenericNode
                                  Index
                                  [ Identifier "mmtm"
                                  , ValueInteger 1
                                  ]
                              , ValueInteger 4
                              ]
                          , ValueInteger 0
                          ]
                      , ValueBoolean True
                      ]
                  , GenericNode
                      Eq
                      [ GenericNode
                          Index
                          [ GenericNode
                              Index
                              [ GenericNode
                                  Index
                                  [ Identifier "mmtm"
                                  , ValueInteger 1
                                  ]
                              , ValueInteger 4
                              ]
                          , ValueInteger 1
                          ]
                      , ValueInteger 3
                      ]
                  , GenericNode
                      Eq
                      [ GenericNode
                          Index
                          [ GenericNode
                              Index
                              [ GenericNode
                                  Index
                                  [ GenericNode
                                      Index
                                      [ Identifier "mmtm"
                                      , ValueInteger 1
                                      ]
                                  , ValueInteger 4
                                  ]
                              , ValueInteger 2
                              ]
                          , ValueInteger 3
                          ]
                      , ValueInteger 12
                      ]
                  ]
  , metadata = []
  }