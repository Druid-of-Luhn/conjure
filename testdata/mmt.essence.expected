language Essence 2.0

find mmt : matrix indexed by [int(1..3), int(4..6)] of tuple of (bool, int(2..6))
find mmtm : matrix indexed by [int(1..3), int(4..6)] of tuple of (bool,
                                                                  int(2..6),
                                                                  matrix indexed by [int(0..9)] of int(10..20))

such that
    mmt[1, 4, 0] = true,
    mmt[1, 4, 1] = 3,
    mmtm[1, 4, 0] = true,
    mmtm[1, 4, 1] = 3,
    mmtm[1, 4, 2, 3] = 12
